src/type/schema.js

```diff
@@ -72,6 +72,62 @@ export function assertSchema(schema: mixed): GraphQLSchema {
  *       mutation: MyAppMutationRootType,
  *     })
  *
+ * Note: When the schema is constructed, only the types that are reachable through
+ * traversing the root types & those that are explicitly referenced are included.
```

Comment:
How about?
> Note: When the schema is constructed by default it includes only types that are reachable through traversing root types by following only explicit references.

Reply:
@IvanGoncharov Does that mean that it will only include types that are mentioned in query field's type or does it mean it will only include those types in fields too only if they are explicitly referenced in the `types` property of schema config?

Reply:
By default, (without `types`) type should be reachable to be included.
Basically, entire schema reduced to the list of types using https://github.com/graphql/graphql-js/blob/bdfd1c63e0b45fc48b9c52c2b652680049d76322/src/type/schema.js#L300
And root types + `config.types` is used as starting set for reduce:
https://github.com/graphql/graphql-js/blob/bdfd1c63e0b45fc48b9c52c2b652680049d76322/src/type/schema.js#L141-L158

Not sure how to comunicate that in clear form without causing even more confusion.

Reply:
@IvanGoncharov How about: 

> Note: When the schema is constructed, by default only the types that are reachable by traversing the root types are included, other types must be explicitly referenced.

Reply:
Looks good! 👍